addons/controllers/webhooks_test.go

```diff
@@ -0,0 +1,155 @@
+			orgKey, err := keyutil.ParsePrivateKeyPEM(testTLSSecret.TLSSecret.Data[resources.ServerKey])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(key).To(Equal(orgKey))
+		})
+	})
+	Context("If the certificates does exists", func() {
+		It("should install new certificates", func() {
+			testTLSSecret := webhooks.WebhookTLS{
+				Ctx:           ctx,
+				K8sConfig:     k8sConfig,
+				CertPath:      certPath,
+				KeyPath:       keyPath,
+				Name:          webhookScrtName,
+				ServiceName:   webhookServiceName,
+				LabelSelector: "webhook-cert=self-managed",
+				Logger:        setupLog,
+				TLSSecret:     nil,
+				Namespace:     addonNamespace,
+				InvalidWithIn: time.Hour * 24,
+			}
+			err := testTLSSecret.Manage()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(err).ToNot(HaveOccurred())
+			Expect(testTLSSecret.TLSSecret).NotTo(BeNil())
+			cert, err := cert2.CertsFromFile(certPath)
+			Expect(err).ToNot(HaveOccurred())
+			certPEM1, err := cert2.EncodeCertificates(cert[0])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(certPEM1).To(Equal(testTLSSecret.TLSSecret.Data[resources.ServerCert]))
+			key, err := keyutil.PrivateKeyFromFile(keyPath)
+			Expect(err).ToNot(HaveOccurred())
+			orgKey, err := keyutil.ParsePrivateKeyPEM(testTLSSecret.TLSSecret.Data[resources.ServerKey])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(key).To(Equal(orgKey))
+
+			err = testTLSSecret.Manage()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(err).ToNot(HaveOccurred())
+			Expect(testTLSSecret.TLSSecret).NotTo(BeNil())
+			cert, err = cert2.CertsFromFile(certPath)
+			Expect(err).ToNot(HaveOccurred())
+			certPEM2, err := cert2.EncodeCertificates(cert[0])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(certPEM2).To(Equal(testTLSSecret.TLSSecret.Data[resources.ServerCert]))
+			key, err = keyutil.PrivateKeyFromFile(keyPath)
+			Expect(err).ToNot(HaveOccurred())
+			orgKey, err = keyutil.ParsePrivateKeyPEM(testTLSSecret.TLSSecret.Data[resources.ServerKey])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(key).To(Equal(orgKey))
+			Expect(certPEM1).To(Equal(certPEM2))
+		})
+	})
+})
+
+var _ = Describe("when TLS certificates are being managed", func() {
+	Context("TLS certificate Management", func() {
+		It("should continuously update and validate certificates ", func() {
+			privCtx := context.Background()
+			cacelCtx, cancelFn := context.WithCancel(privCtx)
+			invalidTime := (time.Hour * 24 * 7) - (time.Second * 10)
+			webhookTLS := webhooks.WebhookTLS{
+				Ctx:           cacelCtx,
+				K8sConfig:     k8sConfig,
+				CertPath:      certPath,
+				KeyPath:       keyPath,
+				Name:          webhookScrtName,
+				ServiceName:   webhookServiceName,
+				LabelSelector: "webhook-cert=self-managed",
+				Logger:        setupLog,
+				TLSSecret:     nil,
+				Namespace:     addonNamespace,
+				InvalidWithIn: invalidTime,
+			}
+
+			go webhooks.ManageTLSCertificates(&webhookTLS, time.Second*10)
+			time.Sleep(time.Second * 10)
```

Comment:
question: why 10 sec ? Would it be possible to add some comments ? I am assuming it is related to the `time.Second*10` passed in on L135, but just a guess. And would it be possible to reduce the time so that tests could run faster ?

Reply:
Reworked the tests to instead of use a explicit time, to use a time derived from the rotation time provided by user. 
Makes test more flexible, and clear.